packages/node/src/indexer/dictionary/v1/substrateDictionaryV1.ts

```diff
@@ -131,7 +131,11 @@ export function buildDictionaryV1QueryEntries<
         filterList = [handler.filter];
       }
       // Filter out any undefined
-      filterList = filterList.filter(Boolean);
+      filterList = filterList.filter(Boolean).map((obj) => {
+        return Object.fromEntries(
+          Object.entries(obj).filter(([key, value]) => value !== undefined),
+        );
+      });
```

Comment:
I think this filtering is better off being on `callFilterToQueryEntry` or `eventFilterToQueryEntry` its very hard to follow whats going on here and using `Object.*` removes a lot of type checks